examples/test_messaging/test_messaging.c

```diff
@@ -0,0 +1,204 @@
+nf_setup(struct onvm_nf_local_ctx *nf_local_ctx);
+
+void
+nf_msg_handler(void *msg_data, struct onvm_nf_local_ctx *nf_local_ctx);
+
+/*
+ * Print a usage message
+ */
+static void
+usage(const char *progname) {
+        printf("Usage:\n");
+        printf("%s [EAL args] -- [NF_LIB args] -- -p <print_delay>\n", progname);
+        printf("%s -F <CONFIG_FILE.json> [EAL args] -- [NF_LIB args] -- [NF args]\n\n", progname);
+        printf("Flags:\n");
+        printf(" - `-p <print_delay>`: number of packets between each print, e.g. `-p 1` prints every packets.\n");
+}
+
+/*
+ * Parse the application arguments.
+ */
+static int
+parse_app_args(int argc, char *argv[], const char *progname) {
+        int c;
+
+        while ((c = getopt(argc, argv, "p:")) != -1) {
+                switch (c) {
+                        case 'p':
+                                print_delay = strtoul(optarg, NULL, 10);
+                                RTE_LOG(INFO, APP, "print_delay = %d\n", print_delay);
+                                break;
+                        case '?':
+                                usage(progname);
+                                if (optopt == 'p')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                else if (isprint(optopt))
+                                        RTE_LOG(INFO, APP, "Unknown option `-%c'.\n", optopt);
+                                else
+                                        RTE_LOG(INFO, APP, "Unknown option character `\\x%x'.\n", optopt);
+                                return -1;
+                        default:
+                                usage(progname);
+                                return -1;
+                }
+        }
+        return optind;
+}
+
+void
+nf_setup(__attribute__((unused)) struct onvm_nf_local_ctx *nf_local_ctx) {
+        struct rte_mempool *pktmbuf_pool;
+
+        pktmbuf_pool = rte_mempool_lookup(PKTMBUF_POOL_NAME);
+        if (pktmbuf_pool == NULL) {
+                onvm_nflib_stop(nf_local_ctx);
+                rte_exit(EXIT_FAILURE, "Cannot find mbuf pool!\n");
+        }
+
+        uint16_t address = nf_local_ctx->nf->service_id;
+
+        int ret = onvm_nflib_send_msg_to_nf(address, NULL);
+        printf("Return: %d\n", ret);
+
+}
+
+void
+nf_msg_handler(void *msg_data, __attribute__((unused)) struct onvm_nf_local_ctx *nf_local_ctx){
+
+        uint16_t address = nf_local_ctx->nf->service_id;
+
+        for(int i = 0; i < 150; i++){
+                int ret = onvm_nflib_send_msg_to_nf(address, msg_data);
+                
+                printf("Return: %d\n", ret);
+                printf("Iteration: %d\n", i);
+                printf("msg_data: %p\n", msg_data);
+                printf("SID: %i\n", address);
+                printf("Ring Count: %u\n", rte_ring_count(nf_local_ctx->nf->msg_q));
+
+                printf("---------------------------\n");
+
+                sleep(1);
```

Comment:
I recommend removing the sleep, and moving the print statements out of the loop and keeping track of how many messages were successfully/unsuccessfully sent. That way we would avoid spamming the console, and give the user a concise conclusion of how the unit test went.

Reply:
Will do. Had that there for my own purposes, but will be gone for final version,